lib/luasocket.c

```diff
@@ -56,15 +56,32 @@ typedef struct luasocket_addr_s {
 static int luasocket_new(lua_State *L);
 static int luasocket_accept(lua_State *L);
 
+static int luasocket_checkiface(lua_State *L, int ix)
+{
+	int ifindex;
+	const char *ifname = luaL_checklstring(L, ix, NULL);
+	struct net_device *dev = dev_get_by_name(&init_net, ifname);
+	if (!dev) {
+		lua_pushinteger(L, -ENODEV);
+		lua_error(L);
+	}
+	ifindex = dev->ifindex;
+	dev_put(dev);
+	return ifindex;
+}
```

Comment:
perhaps this should be added as a Lua binding.. ideally in a `net` library, but could be put temporarily on `linux` lib.. in this way it could be used to other purposes as well.. and in this case we could do something like..
```lua
local ifindex = linux.ifindex("eth0")
sock:send("hello", ifindex)
```

what do you think? btw, I didn't like much the name `ifindex()` for the API, but I couldn't think on something better..

@sheharyaar would be good to hear your thoughts here as well..